### PR TITLE
Add toString for MethodDescriptor and ServiceDescriptor

### PR DESCRIPTION
**Pull request by rmichela**
Implements #2472

* Added `MethodDescriptor.toString()` method.
* Added `ServiceDescriptor.toString()` method.
* Upgraded deprecated code in unit tests.

### CODE REVIEW

File: core/src/main/java/io/grpc/ServiceDescriptor.java [hunk 1]
```diff
@@ -204,4 +205,19 @@ public ServiceDescriptor build() {
...
+  @Override
+  public String toString() {
+    String toString =  MoreObjects.toStringHelper(this)
+        .add("name", name)
+        .add("schemaDescriptor", schemaDescriptor)
+        .omitNullValues()
+        .toString();
+
+    for (MethodDescriptor<?, ?> methodDescriptor : methods) {
+      toString += "\n\t" + methodDescriptor.toString();
```
Comment (zpencer) [inline core/src/main/java/io/grpc/ServiceDescriptor.java]: I looked at some of the other `toString` methods in gRPC and they all seem to be a single line. I'd prefer to keep this a single line as well and optimize for grepping at the cost of human readability.

**Comment by zpencer** [comment]
LGMT with a minor nit.

**Comment by carl-mastrangelo** [comment]
Oh yeah, please set the commit message in our standard format.

**Comment by rmichela** [comment]
Fixed commit message per CONTRIBUTING.md

**Comment by rmichela** [comment]
Does anything else need to be done to merge this?

**Pull request merged**